Doc/library/unittest.mock.rst

```diff
@@ -2831,3 +2831,122 @@ Sealing mocks
         >>> mock.not_submock.attribute2  # This won't raise.
 
     .. versionadded:: 3.7
+
+
+Order of precedence of side_effect, return_value and wraps
+----------------------------------------------------------
+
+The order of their precedence is:
+
+:attr:`~Mock.side_effect` > :attr:`~Mock.return_value` > *wraps*
+
+If all three are set, mock will return the value from :attr:`~Mock.side_effect`
+ignoring :attr:`~Mock.return_value` and the wrapped object altogether. If any
+two are set, the one with the higher precedence will return the value.
+Regardless of the order of which was set first, the order of precedence
+remains unchanged.
+
+    >>> from unittest.mock import Mock
+    >>> class Order:
+    ...     @staticmethod
+    ...     def get_value():
+    ...         return "third"
+    ...
+    >>> order_mock = Mock(spec=Order, wraps=Order)
+    >>> order_mock.get_value.side_effect = ["first"]
+    >>> order_mock.get_value.return_value = "second"
+    >>> order_mock.get_value()
+    'first'
+
+As ``None`` is the default value of :attr:`~Mock.side_effect`, if you reassign
+its value back to ``None``, the order of precedence will be checked between
+:attr:`~Mock.return_value` and the wrapped object, ignoring
+:attr:`~Mock.side_effect`.
+
+    >>> order_mock.get_value.side_effect = None
+    >>> order_mock.get_value()
+    'second'
+
+If the value being returned by :attr:`~Mock.side_effect` is :data:`DEFAULT`,
+it is ignored and the order of precedence moves to the successor to obtain the
+value to return.
+
+    >>> from unittest.mock import DEFAULT
+    >>> order_mock.get_value.side_effect = [DEFAULT]
+    >>> order_mock.get_value()
+    'second'
+
+Unless explicitly set, the default value of :attr:`~Mock.return_value` is
+:data:`DEFAULT`.
+
+    >>> order_mock = Mock(spec=Order, wraps=Order)
+    >>> order_mock.return_value
+    sentinel.DEFAULT
+    >>> order_mock.get_value.return_value
+    sentinel.DEFAULT
+
+Reassigning back to this value or not explicitly assigning any value to it,
+the order of precedence will ignore it and move to the last successor which is
+the wrapped object.
```

Comment:
Not sure this paragraph is necessary.

Reply:
I have removed it but with a slight change.

https://github.com/python/cpython/blob/49a08dc4805a4c0c6ec1095964b9dde89fdf6f8c/Doc/library/unittest.mock.rst#L2881-L2891

```diff
+ When :class:`Mock` wraps an object, the default value of 
+ :attr:`~Mock.return_value` will be :data:`DEFAULT`.
```

This informtion can be verified from this test case 👇:
https://github.com/python/cpython/blob/49a08dc4805a4c0c6ec1095964b9dde89fdf6f8c/Lib/test/test_unittest/testmock/testmock.py#L680-L691

> _It will not be "sentinel.DEFAULT" if the mock is not wrapping any object._